velox/common/memory/Memory.h

```diff
@@ -65,89 +65,174 @@ constexpr uint16_t kDefaultAlignment = 64;
+  /// 'cap'.
+  virtual std::shared_ptr<MemoryPool> addChild(
+      const std::string& name,
+      int64_t cap = kMaxMemory);
 
+  /// Invoked to allocate a buffer with specified 'size'.
   virtual void* FOLLY_NULLABLE allocate(int64_t size) = 0;
+
+  /// Invoked to allocate a zero-filled buffer with capacity that can store
+  /// 'numMembers' entries with each size of 'sizeEach'.
   virtual void* FOLLY_NULLABLE
   allocateZeroFilled(int64_t numMembers, int64_t sizeEach) = 0;
+
+  /// Invoked to re-allocate from an existing buffer with 'newSize' and update
+  /// memory usage counting accordingly. If 'newSize' is larger than the current
+  /// buffer 'size', the function will allocate a new buffer and free the old
+  /// buffer.
   virtual void* FOLLY_NULLABLE
   reallocate(void* FOLLY_NULLABLE p, int64_t size, int64_t newSize) = 0;
+
+  /// Invoked to free an allocated buffer.
   virtual void free(void* FOLLY_NULLABLE p, int64_t size) = 0;
-  virtual size_t getPreferredSize(size_t size) = 0;
 
-  // Accounting/governing methods, lazily managed and expensive for intermediate
-  // tree nodes, cheap for leaf nodes.
+  /// Rounds up to a power of 2 >= size, or to a size halfway between
+  /// two consecutive powers of two, i.e 8, 12, 16, 24, 32, .... This
+  /// coincides with JEMalloc size classes.
+  virtual size_t getPreferredSize(size_t size);
+
+  /// Returns the memory allocation alignment size applied internally by this
+  /// memory pool object.
+  virtual uint16_t getAlignment() const = 0;
+
+  /// Resource governing methods used to track and limit the memory usage
+  /// through this memory pool object.
+
+  /// Accounting/governing methods, lazily managed and expensive for
+  /// intermediate tree nodes, cheap for leaf nodes.
   virtual int64_t getCurrentBytes() const = 0;
+
+  /// Returns the peak memory usage of this memory pool.
   virtual int64_t getMaxBytes() const = 0;
-  // Tracks memory usage from leaf nodes to root and updates immediately
-  // with atomic operations.
-  // Unlike pool's getCurrentBytes(), getMaxBytes(), trace's API's returns
-  // the aggregated usage of subtree. See 'ScopedChildUsageTest' for
-  // difference in their behavior.
+
+  /// Tracks memory usage from leaf nodes to root and updates immediately
+  /// with atomic operations.
+  /// Unlike pool's getCurrentBytes(), getMaxBytes(), trace's API's returns
+  /// the aggregated usage of subtree. See 'ScopedChildUsageTest' for
+  /// difference in their behavior.
   virtual void setMemoryUsageTracker(
       const std::shared_ptr<MemoryUsageTracker>& tracker) = 0;
+
+  /// Returns the memory usage tracker associated with this memory pool.
   virtual const std::shared_ptr<MemoryUsageTracker>& getMemoryUsageTracker()
       const = 0;
-  // Used for external aggregation.
+
+  /// Used for external aggregation.
   virtual void setSubtreeMemoryUsage(int64_t size) = 0;
+
   virtual int64_t updateSubtreeMemoryUsage(int64_t size) = 0;
```

Comment:
What's the difference between setSubtreeMemoryUsage and updateSubtreeMemoryUsage?